include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -70,125 +95,302 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         int m_nknots;
 
     public:
+        /// @brief The type of the discrete dimension representing the B-splines.
         using discrete_dimension_type = NonUniformBSplines;
 
+        /// @brief The type of a DiscreteDomain whose elements identify the B-splines.
         using discrete_domain_type = DiscreteDomain<DDim>;
 
+        /// @brief The type of a DiscreteElement identifying a B-spline.
         using discrete_element_type = DiscreteElement<DDim>;
 
+        /// @brief The type of a DiscreteVector representing an "index displacement" between two B-splines.
         using discrete_vector_type = DiscreteVector<DDim>;
 
         Impl() = default;
 
-        template <class OriginMemorySpace>
-        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
-            : m_domain(impl.m_domain)
-            , m_nknots(impl.m_nknots)
-        {
-        }
-
-        /// @brief Construct a `Impl` using a brace-list, i.e. `Impl bsplines({0., 1.})`
+        /** @brief Constructs an Impl using a brace-list, i.e. `Impl bsplines({0., 1.})`
+         *
+         * Constructs an Impl by iterating over a list of break points. Internally this constructor calls the constructor
+         * Impl(RandomIt breaks_begin, RandomIt breaks_end).
+         *
+         * @param breaks The std::initializer_list of the coordinates of break points.
+         */
         explicit Impl(std::initializer_list<ddc::Coordinate<Tag>> breaks)
             : Impl(breaks.begin(), breaks.end())
         {
         }
 
-        /// @brief Construct a `Impl` using a C++20 "common range".
+        /** @brief Constructs an Impl using a std::vector.
+         *
+         * Constructs an Impl by iterating over a list of break points. Internally this constructor calls the constructor
+         * Impl(RandomIt breaks_begin, RandomIt breaks_end).
+         *
+         * @param breaks The std::vector of the coordinates of break points.
+         */
         explicit Impl(std::vector<ddc::Coordinate<Tag>> const& breaks)
             : Impl(breaks.begin(), breaks.end())
         {
         }
 
-        /// @brief Construct a `Impl` using a pair of iterators.
+        /** @brief Constructs an Impl by iterating over a set of break points from begin to end.
+         *
+         * The provided break points describe the separation between the cells on which the polynomials
+         * comprising a spline are defined. They are used to build a set of knots. There are 2*degree more
+         * knots than break points. The knots are defined as follows:
+         * \f$ k_i = b_0 \forall 0 \leq i < d \f$
+         * \f$ k_{i+d} = b_i \forall 0 \leq i < n_b \f$
+         * \f$ k_{i+d+n_b} = b_{n_b} \forall 0 \leq i < d \f$
```

Comment:
```suggestion
         * \f$ k_{i+d+n_b} = b_{n_b-1} \forall 0 \leq i < d \f$
```
Just realised that there is an index problem here